json.go

```diff
@@ -0,0 +1,324 @@
+		err := iter.Error
+		switch l1Field {
+		case "apiVersion":
+			err = static.setGroupVersionFromAPI(iter.ReadString())
+
+		case "kind":
+			static.Kind = iter.ReadString()
+			if err == nil && builder.SetGroupVersionKind != nil {
+				err = builder.SetGroupVersionKind(static.Group, static.Version, static.Kind)
+			}
+
+		case "metadata":
+			common := CommonMetadata{}
+			for l2Field := iter.ReadObject(); l2Field != ""; l2Field = iter.ReadObject() {
+				switch l2Field {
+				case "namespace":
+					static.Namespace = iter.ReadString()
+				case "name":
+					static.Name = iter.ReadString()
+				case "resourceVersion":
+					common.ResourceVersion = iter.ReadString()
+				case "uid":
+					common.UID = iter.ReadString()
+				case "creationTimestamp":
+					t, err := time.Parse(time.RFC3339, iter.ReadString())
+					if err != nil {
+						return fmt.Errorf("invalid updatedTimestamp format // %w", err)
+					}
+					common.CreationTimestamp = t
+				case "finalizers":
+					common.Finalizers = make([]string, 0)
+					for iter.ReadArray() {
+						common.Finalizers = append(common.Finalizers, iter.ReadString())
+					}
+
+				case "annotations":
+					for anno := iter.ReadObject(); anno != ""; anno = iter.ReadObject() {
+						val := iter.ReadString()
+
+						i := strings.Index(anno, "/")
+						if i > 0 {
+							g := anno[:i]
+							v := anno[i+1:]
+
+							if g == "grafana.com" {
+								switch v {
+								case "createdBy":
+									common.CreatedBy = val
+								case "updatedBy":
+									common.UpdatedBy = val
+								case "creationTimestamp":
+									t, err := time.Parse(time.RFC3339, val)
+									if err != nil {
+										return fmt.Errorf("invalid creationTimestamp format // %w", err)
+									}
+									common.CreationTimestamp = t
+								case "updatedTimestamp":
+									t, err := time.Parse(time.RFC3339, val)
+									if err != nil {
+										return fmt.Errorf("invalid updatedTimestamp format // %w", err)
+									}
+									common.UpdateTimestamp = t
+								case "origin.name":
+									if common.Origin == nil {
+										common.Origin = &ResourceOriginInfo{}
+									}
+									common.Origin.Name = val
+								case "origin.path":
+									if common.Origin == nil {
+										common.Origin = &ResourceOriginInfo{}
+									}
+									common.Origin.Path = val
+								case "origin.key":
+									if common.Origin == nil {
+										common.Origin = &ResourceOriginInfo{}
+									}
+									common.Origin.Key = val
+								case "origin.timestamp":
+									if common.Origin == nil {
+										common.Origin = &ResourceOriginInfo{}
+									}
+									t, err := time.Parse(time.RFC3339, val)
+									if err != nil {
+										return fmt.Errorf("invalid updatedTimestamp format // %w", err)
+									}
+									common.Origin.Timestamp = &t
+								default:
+									fmt.Printf("grafana anno> %s = %v\n", g, v)
+								}
+							} else {
```

Comment:
For non-`grafana.com/*` annotations, we'll want to add them to the annotations in `ExtraFields`. I think in the future we'll also just want `Annotations` exposed in `CommonMetadata` anyway.